lib/utils/search.js

```diff
@@ -20,29 +18,61 @@ const getSearchResults = (searchTerm, items) => {
   });
 };
 
-const getWeight = (searchTerm, words, resourceIndex, synonyms) => {
+const getWeight = (searchTerm, words, resource, resourceIndex) => {
+  let weight = 0;
+
+  weight += getFullPhraseWeight(searchTerm, resourceIndex);
+  weight += getIndividualWordsWeight(words, resourceIndex);
+  weight += getSynonymsWeight(words, resourceIndex, searchTerm);
+
+  const canReferToService = resource.referralContact || resource.referralWebsite;
+
+  if (weight > 0 && canReferToService) weight += 50;
+
+  return weight;
+};
+
+const getFullPhraseWeight = (searchTerm, resourceIndex) => {
+  if (resourceIndex.some(x => x.includes(searchTerm))) return 75;
+  return 0;
+};
+
+const getIndividualWordsWeight = (words, resourceIndex) => {
+  let wordsMatchedInTitle = words.filter(word => resourceIndex[0].includes(word)).length;
+  let wordsMatched = words.filter(word => resourceIndex.some(x => x.includes(word))).length;
+
+  return wordsMatched * 25 + wordsMatchedInTitle * 25;
+};
+
+const getSynonymsWeight = (words, resourceIndex, searchTerm) => {
+  const synonyms = isJSON(process.env.SYNONYMS) ? JSON.parse(process.env.SYNONYMS) : '{}';
+
+  const foundSynonymsFullPhrase = Object.keys(synonyms).filter(key =>
+    synonyms[key].includes(searchTerm)
+  );
+
   const foundSynonyms = words.map(word =>
     Object.keys(synonyms).filter(key => synonyms[key].includes(word))
   );
```

Comment:
Maybe we can move this before the first return (line 63), cause we only want to run this if full phrase isn't matched

Reply:
Good spot, sorted!